### PR TITLE
feat: add ci for prettier check and build

### LINKED ISSUE #47: Add CI
Would be great to setup github actions to run prettier --check and a build on each PR to ensure those are passing

### PR DESCRIPTION
**Pull request by kunal00000**
Added CI for prettier check and build on PRs and pushes.
Resolves #47

### CODE REVIEW

**Review by ManindraDeMel — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '20'
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run prettier
+        run: npm run prettier:check
```
Comment (marcelovicentegc) [inline .github/workflows/build.yml:21]: Static tests could run on git hooks/pre-commits as well along with a linter 😄

File: package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
     "crawlee": "^3.0.0",
     "glob": "^10.3.10",
     "inquirer": "^9.2.12",
-    "playwright": "*"
+    "playwright": "*",
+    "prettier": "^3.1.0"
```
[the remark this answers is not in the document]
Comment (kunal00000) [reply package.json:15]: Yeah sure.

**Comment by kunal00000** [comment]
I gave rebasing a shot, and I'm crossing my fingers that I got it right. 🧐 I always feel a bit on edge when I'm rebasing branches, especially when there are conflicts involved.

**Comment by steve8708** [comment]
apologies, we just merged a few more things, needs one more round of conflict resolution and then we can get this in!

**Review by marcelovicentegc — CHANGES_REQUESTED** [review]
I'm leaving a suggestion to open a new PR including only the changes for the prettier check step on CI since this PR duplicates the git history (per https://github.com/BuilderIO/gpt-crawler/pull/48#issuecomment-1828097212)

**Pull request closed without merging**